llvm/test/CodeGen/CHERI-Generic/RISCV64/purecap-jumptable.ll

```diff
@@ -98,47 +98,48 @@ exit:
 define void @above_threshold_all(i32 %in, i32 addrspace(200)* %out) nounwind {
 ; CHECK-LABEL: above_threshold_all:
 ; CHECK:       # %bb.0: # %entry
+; CHECK-NEXT:    cincoffset csp, csp, -16
+; CHECK-NEXT:    csc cra, 0(csp) # 16-byte Folded Spill
 ; CHECK-NEXT:    sext.w a0, a0
-; CHECK-NEXT:    addi a2, zero, 3
-; CHECK-NEXT:    blt a2, a0, .LBB2_5
-; CHECK-NEXT:  # %bb.1: # %entry
-; CHECK-NEXT:    addi a2, zero, 1
-; CHECK-NEXT:    beq a0, a2, .LBB2_9
-; CHECK-NEXT:  # %bb.2: # %entry
-; CHECK-NEXT:    addi a2, zero, 2
-; CHECK-NEXT:    beq a0, a2, .LBB2_10
-; CHECK-NEXT:  # %bb.3: # %entry
-; CHECK-NEXT:    addi a2, zero, 3
-; CHECK-NEXT:    bne a0, a2, .LBB2_14
-; CHECK-NEXT:  # %bb.4: # %bb3
-; CHECK-NEXT:    addi a0, zero, 2
-; CHECK-NEXT:    j .LBB2_13
-; CHECK-NEXT:  .LBB2_5: # %entry
-; CHECK-NEXT:    addi a2, zero, 4
-; CHECK-NEXT:    beq a0, a2, .LBB2_11
-; CHECK-NEXT:  # %bb.6: # %entry
+; CHECK-NEXT:    addi a0, a0, -1
 ; CHECK-NEXT:    addi a2, zero, 5
-; CHECK-NEXT:    beq a0, a2, .LBB2_12
-; CHECK-NEXT:  # %bb.7: # %entry
-; CHECK-NEXT:    addi a2, zero, 6
-; CHECK-NEXT:    bne a0, a2, .LBB2_14
-; CHECK-NEXT:  # %bb.8: # %bb6
-; CHECK-NEXT:    addi a0, zero, 200
-; CHECK-NEXT:    j .LBB2_13
-; CHECK-NEXT:  .LBB2_9: # %bb1
+; CHECK-NEXT:    bltu a2, a0, .LBB2_9
+; CHECK-NEXT:  # %bb.1: # %entry
+; CHECK-NEXT:    slli a0, a0, 2
+; CHECK-NEXT:  .LBB2_10: # %entry
+; CHECK-NEXT:    # Label of block must be emitted
+; CHECK-NEXT:    auipcc ca2, %pcrel_hi(.LJTI2_0)
+; CHECK-NEXT:    cincoffset ca2, ca2, %pcrel_lo(.LBB2_10)
+; CHECK-NEXT:    cincoffset ca0, ca2, a0
+; CHECK-NEXT:    clw a0, 0(ca0)
```

Comment:
Checking the contents of the jump table with `UTC_ARGS: --disable` could also be useful to ensure they actually look sensible.

Reply:
I wanted to autogenerate it like update_cc_test_checks.py can for globals but then realised update_llc_test_checks.py can't do that. I guess I could just manually writing the CHECK lines like you suggest, that's probably a good idea.